src/engine/platform/scc.cpp

```diff
@@ -0,0 +1,387 @@
+  "Ch1_Vol", "8a",
+  "Ch2_Vol", "8b",
+  "Ch3_Vol", "8c",
+  "Ch4_Vol", "8d",
+  "Ch5_Vol", "8e",
+  "Output", "8f",
+  "Test", "e0",
+  NULL
+};
+
+const char* regCheatSheetSCCPlus[]={
+  "Ch1_Wave", "00",
+  "Ch2_Wave", "20",
+  "Ch3_Wave", "40",
+  "Ch4_Wave", "60",
+  "Ch5_Wave", "80",
+  "Ch1_FreqL", "a0",
+  "Ch1_FreqH", "a1",
+  "Ch2_FreqL", "a2",
+  "Ch2_FreqH", "a3",
+  "Ch3_FreqL", "a4",
+  "Ch3_FreqH", "a5",
+  "Ch4_FreqL", "a6",
+  "Ch4_FreqH", "a7",
+  "Ch5_FreqL", "a8",
+  "Ch5_FreqH", "a9",
+  "Ch1_Vol", "aa",
+  "Ch2_Vol", "ab",
+  "Ch3_Vol", "ac",
+  "Ch4_Vol", "ad",
+  "Ch5_Vol", "ae",
+  "Output", "af",
+  "Test", "c0",
+  NULL
+};
+
+const char** DivPlatformSCC::getRegisterSheet() {
+  return isPlus ? regCheatSheetSCCPlus : regCheatSheetSCC;
+}
+
+const char* DivPlatformSCC::getEffectName(unsigned char effect) {
+  switch (effect) {
+    case 0x10:
+      return "10xx: Change waveform";
+      break;
+  }
+  return NULL;
+}
+
+void DivPlatformSCC::acquire(short* bufL, short* bufR, size_t start, size_t len) {
+  for (size_t h=start; h<start+len; h++) {
+    for (int i=0; i<16; i++) {
+      scc->tick();
+    }
+    short out=(short)scc->out()<<5;
+    bufL[h]=bufR[h]=out;
+
+    for (int i=0; i<5; i++) {
+      oscBuf[i]->data[oscBuf[i]->needle++]=scc->chan_out(i)<<7;
+    }
+  }
+}
+
+void DivPlatformSCC::updateWave(int ch) {
+  int dstCh=(!isPlus && ch>=4)?3:ch;
+  for (int i=0; i<32; i++) {
+    rWrite(dstCh*32+i,(unsigned char)chan[ch].ws.output[i]-128);
+  }
+}
+
+void DivPlatformSCC::tick(bool sysTick) {
+  for (int i=0; i<5; i++) {
+    chan[i].std.next();
+    if (chan[i].std.vol.had) {
+      chan[i].outVol=((chan[i].vol&15)*MIN(15,chan[i].std.vol.val))/15;
+      rWrite(regBase+10+i,chan[i].outVol);
+    }
+    if (chan[i].std.arp.had) {
+      if (!chan[i].inPorta) {
+        if (chan[i].std.arp.mode) {
+          chan[i].baseFreq=NOTE_PERIODIC(chan[i].std.arp.val);
+        } else {
+          chan[i].baseFreq=NOTE_PERIODIC(chan[i].note+chan[i].std.arp.val);
+        }
+      }
+      chan[i].freqChanged=true;
+    } else {
+      if (chan[i].std.arp.mode && chan[i].std.arp.finished) {
+        chan[i].baseFreq=NOTE_PERIODIC(chan[i].note);
+        chan[i].freqChanged=true;
+      }
+    }
+    if (chan[i].std.wave.had) {
+      if (chan[i].wave!=chan[i].std.wave.val || chan[i].ws.activeChanged()) {
+        chan[i].wave=chan[i].std.wave.val;
+        chan[i].ws.changeWave1(chan[i].wave);
+      }
+    }
+    if (chan[i].std.pitch.had) {
+      if (chan[i].std.pitch.mode) {
+        chan[i].pitch2+=chan[i].std.pitch.val;
+        CLAMP_VAR(chan[i].pitch2,-2048,2048);
+      } else {
+        chan[i].pitch2=chan[i].std.pitch.val;
+      }
+      chan[i].freqChanged=true;
+    }
+    if (chan[i].active) {
+      if (chan[i].ws.tick()) {
+        updateWave(i);
+      }
+    }
+    if (chan[i].freqChanged) {
+      chan[i].freq=parent->calcFreq(chan[i].baseFreq,chan[i].pitch,true)+chan[i].pitch2;
```

Comment:
Replace with `chan[i].freq=parent->calcFreq(chan[i].baseFreq,chan[i].pitch,true,0,chan[i].pitch2,chipClock,CHIP_DIVIDER);` to comply with latest frequency calculation changes.